x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/logging/DeprecationIndexingComponent.java

```diff
@@ -51,10 +52,22 @@ public class DeprecationIndexingComponent extends AbstractLifecycleComponent imp
     private static final Logger logger = LogManager.getLogger(DeprecationIndexingComponent.class);
 
     private final DeprecationIndexingAppender appender;
-    private final BulkProcessor processor;
+    private final BulkProcessor2 processor;
+    private final Client client;
+    /*
+     * We do not want to index deprecation logs on server startup before the index template and ILM policy are available. So we have
+     * processor put them in the following buffer until the server is ready. Once the template and ILM policy are available, this buffer is
+     * drained and its contents are sent to the processor. The queue is unbounded because we are first going through processor::add, which
+     * starts rejecting documents if the total number of bytes in flight gets too large.
+     */
+    private final BlockingQueue<Tuple<BulkRequest, ActionListener<BulkResponse>>> requestAndListenerBuffer = new LinkedBlockingQueue<>();
```

Comment:
I'd be inclined to make this a queue of `Runnable`, there's no need to capture the request and listener separately. Also maybe use a `ConcurrentLinkedQueue` which avoids the need to handle any `InterruptedException` since there won't be any blocking here anyway.

Reply:
in the light of https://github.com/elastic/elasticsearch/pull/94211/commits/fa87b0b1899a9db5e2ac87b89b6043d8f7f22e76 maybe a queue of `AbstractRunnable`, that way you can fail any requests that you never got around to even sending. Better that than just leaking them IMO.